features/ui/sidebar-navigation/menu-item-link.tsx

```diff
@@ -33,9 +33,11 @@ export const Anchor = styled.a`
   text-decoration: none;
 `;
 
-export const Icon = styled.img`
+export const Icon = styled.img<{ isCollapsed?: boolean; text?: string }>`
   width: ${space(6)};
   margin-right: ${space(3)};
+  transform: ${({ isCollapsed, text }) =>
+    isCollapsed && text === "Collapse" ? "rotate(180deg)" : ""};
```

Comment:
This CSS is exactly what's required. But to make things easier you could move it to the `sidebar-navigation.tsx` file. You can find a component for the collapse button there iirc.

The problem with this implementation is the tight coupling of the text with the rotation style. What if we change the text? What if we need internationalization and the text is translated? So it's better to decouple this from the text